src/internal/operations/export.go

```diff
@@ -0,0 +1,422 @@
+	bytesRead     *stats.ByteCounter
+	resourceCount int
+
+	// a transient value only used to pair up start-end events.
+	exportID string
+}
+
+// Run begins a synchronous export operation.
+func (op *ExportOperation) Run(ctx context.Context) (
+	expColl []data.ExportCollection,
+	err error,
+) {
+	defer func() {
+		if crErr := crash.Recovery(ctx, recover(), "export"); crErr != nil {
+			err = crErr
+		}
+	}()
+
+	var (
+		opStats = exportStats{
+			bytesRead: &stats.ByteCounter{},
+			exportID:  uuid.NewString(),
+		}
+		start  = time.Now()
+		sstore = streamstore.NewStreamer(op.kopia, op.acct.ID(), op.Selectors.PathService())
+	)
+
+	// -----
+	// Setup
+	// -----
+
+	ctx, end := diagnostics.Span(ctx, "operations:export:run")
+	defer func() {
+		end()
+		// wait for the progress display to clean up
+		observe.Complete()
+	}()
+
+	ctx, flushMetrics := events.NewMetrics(ctx, logger.Writer{Ctx: ctx})
+	defer flushMetrics()
+
+	ctx = clues.Add(
+		ctx,
+		"tenant_id", clues.Hide(op.acct.ID()),
+		"backup_id", op.BackupID,
+		"service", op.Selectors.Service)
+
+	defer func() {
+		op.bus.Event(
+			ctx,
+			events.ExportEnd,
+			map[string]any{
+				events.BackupID:      op.BackupID,
+				events.DataRetrieved: op.Results.BytesRead,
+				events.Duration:      op.Results.CompletedAt.Sub(op.Results.StartedAt),
+				events.EndTime:       dttm.Format(op.Results.CompletedAt),
+				events.ItemsRead:     op.Results.ItemsRead,
+				events.ItemsWritten:  op.Results.ItemsWritten,
+				events.Resources:     op.Results.ResourceOwners,
+				events.ExportID:      opStats.exportID,
+				events.Service:       op.Selectors.Service.String(),
+				events.StartTime:     dttm.Format(op.Results.StartedAt),
+				events.Status:        op.Status.String(),
+			})
+	}()
+
+	// -----
+	// Execution
+	// -----
+
+	expCollections, err := op.do(ctx, &opStats, sstore, start)
+	if err != nil {
+		// No return here!  We continue down to persistResults, even in case of failure.
+		logger.CtxErr(ctx, err).Error("running export")
+
+		if errors.Is(err, kopia.ErrNoRestorePath) {
+			op.Errors.Fail(clues.New("empty backup or unknown path provided"))
+		}
+
+		op.Errors.Fail(clues.Wrap(err, "running export"))
+	}
+
+	finalizeErrorHandling(ctx, op.Options, op.Errors, "running export")
+	LogFaultErrors(ctx, op.Errors.Errors(), "running export")
+
+	// -----
+	// Persistence
+	// -----
+
+	err = op.persistResults(ctx, start, &opStats)
+	if err != nil {
+		op.Errors.Fail(clues.Wrap(err, "persisting export results"))
+		return nil, op.Errors.Failure()
+	}
+
+	logger.Ctx(ctx).Infow("completed export", "results", op.Results)
+
+	return expCollections, nil
+}
+
+type zipExportCol struct {
+	reader io.ReadCloser
+}
+
+func (z zipExportCol) GetBasePath() string {
+	return ""
+}
+
+func (z zipExportCol) GetItems(ctx context.Context) <-chan data.ExportItem {
+	rc := make(chan data.ExportItem, 1)
+	defer close(rc)
+
+	rc <- data.ExportItem{
+		Data: data.ExportItemData{
+			Name: "export.zip",
```

Comment:
Let me change the default name to `"Corso_Export_"+dttm.FormatNow(dttm.HumanReadable)+".zip"`. As for making it configurable, this is merely a suggested name and since there will only be one item to restore, they can just choose to ignore the name. Or are you talking about cli users being able to edit the name of the archive being created.

Related: https://github.com/alcionai/corso/pull/3793#discussion_r1262244736 /cc @ryanfkeepers

Reply:
@ryanfkeepers I'm not sure if I fully understand you. By "when given a file", did you mean in the case where we are archiving only a single file?

Reply:
as in, depending on the user's input.  `--archive /foo/bar/baz`.  If baz is a directory, archive to `.../baz/standard_name`.  Otherwise assume baz is a non-existing file.

Reply:
Hmm, I feel a bit uneasy about that. I feel it kinda makes the operation a bit unpredictable. Any thoughts @gmatev ?